std/typecons.d

```diff
@@ -768,6 +770,16 @@ if (distinctFieldNames!(Specs))
             return field[] == rhs.field[];
         }
 
+        bool opEquals(R...)(const auto ref R rhs) const
```

Comment:
I don't think `const` is a good idea here:

```
struct Test(T...)
{
    T field;
    
    this (T t) { field = t; }
    
    bool opEquals(R...)(const auto ref R rhs) const
    {
        static foreach (i, _; T)
            if (field[i] != rhs[i])
            	return false;
            
        return true;
    }
}

struct Bad
{
    int a;
    
    bool opEquals(Bad b)
    {
        return a == b.a;
    }
}

void main()
{
	import std.meta;
    auto t = Test!(int, Bad, string)(1, Bad(1), "asdf");
    assert(t == AliasSeq!(1, Bad(1), "asdf"));
}
```

Reply:
Urgh. That really sucks. Well, we could duplicate the `opEquals` yet another time, but for now I just removed both `const`'s and added your example to the source code, s.t. we don't regress against this.
Thanks!